### ISSUE TITLE
Use Xcode command line tools instead of Command Line Tools

### ISSUE THREAD

**Issue by cbrnr** [Homebrew/brew#1613]
Is there a specific reason why `brew` requires the Command Line Tools package (in `/Library/Developer/CommandLineTools/usr`)? Since Xcode also provides these tools (in `/Applications/Xcode.app/Contents/Developer/usr`), I was wondering why it is necessary to have both packages installed in case Xcode is installed anyway.

**Comment by ilovezfs** [comment]
>Is there a specific reason why brew requires the Command Line Tools package (in /Library/Developer/CommandLineTools/usr)

There is actually no such requirement. However, there are formulae that will be forced to build from source if you do not have the CLT. They can still be built from source with Xcode only, but because the pre-built bottles are compiled in an environment that has both Xcode and the CLT installed, there are some cases where the bottles end up having a hard dependency on the CLT. A major example is gcc. So installing the CLT may help you avoid some lengthy source builds.

We ensure that all Homebrew formulae can be built with Xcode.app alone. Most formulae can be built with just the CLT, and those that require the full Xcode.app have an explicit`depends_on :xcode => :build`. Some users would prefer to use only the CLT because it's a much smaller download and takes less time to install and upgrade than Xcode.

That said, there are files installed in, for example (there are others), /usr/include that you will only have if you've installed the CLT. So outside of Homebrew you may have an easier time building software that doesn't know it can also find those headers in the Xcode SDK folder. We make sure the formulae work around such problems.

Bottom line: install both if you want to be able to use as many pre-built binary bottles as possible, and want to be able to build formula that have a `depends_on :xcode => :build`.

**Comment by cbrnr** [comment]
@ilovezfs thank you for this detailed answer! Regarding the bit about files in /usr/include that are only there if the CLT are installed - doesn't Mac OS X use shims in this folder which redirect to a real implementation if one is installed? That is, I don't quite understand why any application might need to know that some headers are actually living in the Xcode SDK folder. If they're in Xcode, they're also in /usr/include, or is this assumption incorrect?

Anyway, I understand the implications of installing either or both of these packages now.

**Comment by ilovezfs** [comment]
@cbrnr the file just aren't there in /usr/include if you don't have the CLT. You could probably install a symlink into the Xcode SDK folder but /usr isn't writable when SIP is on so that's not really a viable option.
